v2/git-initializer/main.go

```diff
@@ -102,10 +102,62 @@ func gitCheckout() error {
 		auth = publicKeys
 	}
 
+	var worktree *git.Worktree
+	if event.Worker.Git.Commit == "" && event.Worker.Git.Ref == "" {
+		worktree, err = simpleCheckout(gitConfig.CloneURL, auth)
```

Comment:
If we had no commit and no ref-- absolutely nothing to go on-- then a simple clone gets us what we want-- the default branch, regardless of what's named.